lib/kazan/client/imp.ex

```diff
@@ -44,9 +44,17 @@ defmodule Kazan.Client.Imp do
 
     with {:ok, result} <- res,
          {:ok, body} <- check_status(result),
-         {:ok, data} <- Poison.decode(body),
-         {:ok, model} <- Kazan.Models.decode(data, request.response_schema),
-         do: {:ok, model}
+         {:ok, content_type} <- get_content_type(result) do
+      case content_type do
+        "application/json" ->
+          with {:ok, data} <- Poison.decode(body),
+               {:ok, model} <- Kazan.Models.decode(data, request.response_schema),
+               do: {:ok, model}
+
+        _ ->
+          {:ok, body}
```

Comment:
I'm not sure about this being a catch-all case.  I'm wondering if it'd be better to explicitly return `{:ok, body}` for `text/plain` responses, but error on other responses...

Reply:
Agreed - Fixed